snuba/datasets/entities/sessions.py

```diff
@@ -211,8 +212,24 @@ def __init__(self) -> None:
     def select_storage(
         self, query: Query, request_settings: RequestSettings
     ) -> StorageAndMappers:
-        granularity = extract_granularity_from_query(query, "started") or 3600
-        use_materialized_storage = granularity >= 3600 and (granularity % 3600) == 0
+        use_materialized_storage: Optional[bool] = None
+
+        # If the passed in `request_settings` arg is an instance of `SubscriptionRequestSettings`,
+        # then it is a crash rate alert subscription, and hence we decide on whether to use the
+        # materialized storage or the raw storage by examining the time_window.
+        # If the `time_window` <=1h, then select the raw storage otherwise select materialized
+        # storage
+        # NOTE: If we were to support other types of subscriptions over the sessions dataset that
+        # do not follow this method used to identify which storage to use, we would need to
+        # match functions used with a regex.
+        if isinstance(request_settings, SubscriptionRequestSettings):
+            from_date, to_date = get_time_range(query, "started")  # type: ignore
```

Comment:
Why did you add `# type: ignore`?

Reply:
mypy was failing because this query is not an instance of `ProcessableQuery[Table]` .. I could alternatively cast the type if it makes more sense?

Reply:
We shouldn't ignore or cast it to the wrong type. `get_time_range` currently expects to be called with `ProcessableQuery[Table]` which is a type of `Query` later in the processing pipeline. If that function is expected to work with any kind of query we can change the type of the `get_time_range` function to allow that.